### DISCUSSION TITLE
I received an error while compiling the A2DP Sink (Music Receiver) code

### DISCUSSION THREAD

**Discussion by syedadil19** [pschatzmann/ESP32-A2DP#498] [Q&A] [answered]
Hi, I've been trying to implement a Bluetooth speaker connected to a mobile phone. I was just starting doing it as I've just received my ESP32 Dev Module. 

When I compiled the following code,
#include "AudioTools.h"
#include "BluetoothA2DPSink.h"

I2SStream out;
BluetoothA2DPSink a2dp_sink(out);

void setup() {
    auto cfg = out.defaultConfig();
    config.pin_bck = 26;
    config.pin_ws = 25;
    config.pin_data = 22;

    out.begin(cfg); 
    a2dp_sink.start("MyMusic");
}

void loop() {
    delay(100);
}

I received the errors as listed by Arduino IDE:

In file included from /Users/syedmohamadadil/Downloads/PWM_generator/PWM_generator.ino:2:
/Users/syedmohamadadil/Documents/Arduino/libraries/ESP32-A2DP/src/BluetoothA2DPSink.h: In constructor 'BluetoothA2DPSink::BluetoothA2DPSink()':
/Users/syedmohamadadil/Documents/Arduino/libraries/ESP32-A2DP/src/BluetoothA2DPSink.h:113:9: error: 'is_i2s_active' was not declared in this scope
         is_i2s_active = false;
         ^~~~~~~~~~~~~
/Users/syedmohamadadil/Documents/Arduino/libraries/ESP32-A2DP/src/BluetoothA2DPSink.h:113:9: note: suggested alternative: 'set_i2s_active'
         is_i2s_active = false;
         ^~~~~~~~~~~~~
         set_i2s_active
/Users/syedmohamadadil/Downloads/PWM_generator/PWM_generator.ino: In function 'void setup()':
/Users/syedmohamadadil/Downloads/PWM_generator/PWM_generator.ino:9:5: error: 'config' was not declared in this scope
     config.pin_bck = 26;
     ^~~~~~
/Users/syedmohamadadil/Downloads/PWM_generator/PWM_generator.ino:9:5: note: suggested alternative: 'cfg'
     config.pin_bck = 26;
     ^~~~~~
     cfg

exit status 1

Compilation error: 'config' was not declared in this scope

I've downloaded and added the zip libraries (both the A2DP and Audio Tools) to Arduino IDE library so I'm not sure what is happening. Do anyone mind to help me with what's going on here? Thank you in advance.

**Comment by pschatzmann** [comment, marked as answer]
Thanks for letting me know. It seems that the last merge went bad. 
I have deleted it and it should be working again...

**Reply by syedadil19** [reply]
Does that mean I'll need to download the zip file again?

**Comment by pschatzmann** [comment]
If you did not use git, then yes: otherwise you would just execute git pull to get the latest changes

**Reply by syedadil19** [reply]
I’ve downloaded and added the zip file for a2dp again and compiled the latest code you posted. It’s compiling and uploading without errors now. Thank you for your quick response.
